src/System.Linq.Expressions/tests/ExceptionHandling/ExceptionHandlingExpressions.cs

```diff
@@ -527,6 +527,258 @@ public void FilterOnCatch(bool useInterpreter)
         }
 
         [Fact]
+        [ActiveIssue(3838)]
+        public void FilterOnCatchCompiled(bool useInterpreter)
+        {
+            FilterOnCatch(false);
+        }
+
+        [Theory]
+        [InlineData(true)]
+        public void FilterCanAccessException(bool useInterpreter)
+        {
```

Comment:
Need a test where filter writes to the exception parameter (assign null or something).
In CIL that should not affect the exception object pushed for the handler since they are different values.

I think we should have correct behavior here, but need test to make sure it is correct and does not regress. It is kind of subtle.

Especially that in C# assignment in filter is visible in the handler because C# models exception object as a single variable, where in CIL they are values.

Here is a C# sample to illustrate the scenario, although I believe we need CIL semantics here not C#
```C#
            try
            {
                throw new NullReferenceException();
            }
            catch (Exception ex) when ((ex = null) == null)
            {
                // C# ignores the ex value re-pushed by IL for the handler
                // since in C# ex is a variable.
                System.Console.WriteLine(ex == null);
            }
```